File: src/jray/ray/Scene.java
package jray.ray;

import java.util.List;

import jray.common.Object3D;
import jray.common.Sky;
import jray.struct.Octree;

abstract public class Scene {

    protected String name = this.getClass().getName();

    public abstract Camera getCamera();
    public abstract Sky getSky();

    public abstract List<Object3D> getObjects();

    public String getName() {
        return name;
    }
    
    public Octree getSceneTree(){
    	return null;
    }
}

File: JRayXLib/JRayXLib/Ray/Scene.cs
using System;
using JRayXLib.Shapes;
using JRayXLib.Struct;

namespace JRayXLib.Ray
{
    public abstract class Scene
    {
        protected string Name;

        protected Scene()
        {
            Name = GetType().Name;
            LightDirection = new Vect3(0, -1, .5).Normalize();
        }

        public I3DObject[] Objects { get; set; }
        public Camera Camera { get; set; }
        public Vect3 LightDirection { get; set; }

        public abstract Sky GetSky();
        public abstract Octree GetSceneTree();

        public String GetName()
        {
            return Name;
        }
    }
}